### PR TITLE
Don't generate a transaction for a details_for call

### PR DESCRIPTION
**Pull request by alenad**

### CODE REVIEW

**Comment by alenad** [comment]
@pierre @daliwei

**Comment by pierre** [comment]
Could you add a test case to verify https://github.com/killbill/killbill-paypal-express-plugin/issues/26 is indeed fixed with this patch?

**Comment by pierre** [comment]
For the testing, you need to verify `@plugin.get_payment_info` returns a single payment `PaymentTransactionInfoPlugin` per payment transaction (for a typical auth/capture, only two should be returned).

What needs to be checked:

* Plugin status is set correctly on a per transaction basis (e.g. `ERROR` for failed transactions)
* Plugin properties returned are the ones from the actual payment API call, not the `details_for` one. You expressed interest in merging the two, i.e. augmenting the properties from the payment API call with the `details_for` ones to return additional information like `payerEmail`. This is absolutely fine but for conflicting ones (`message`, `success`, etc.), the ones from the payment API call take precedence.
* Verify that the `PaymentTransactionInfoPlugin` returned by the initial API call (e.g. `authorizePayment`) and the one from `getPaymentInfo` are consistent (it isn't anymore in case of failure, which can confuse Kill Bill for `PENDING` and `UNKNOWN`).
* I don't see many negative tests (insufficient funds, account closed, etc.) besides a double capture. It would be worth adding a new negative scenario (e.g. failed auth) since this issue mainly manifests itself for failed payments.

Also, all existing tests need to be updated, as they will probably fail with your patch (see https://github.com/killbill/killbill-paypal-express-plugin/blob/66c7d2bc3c4ed6c4efa2a9eed57aba959c814a93/spec/paypal_express/remote/hpp_spec_helpers.rb#L58 for example).

@daliwei has credentials to run the full suite (`bundle exec rake test:remote:spec`) if you don't have any.

**Comment by alenad** [comment]
@pierre updated and tested this PR

File: lib/paypal_express/api.rb [hunk 1]
```diff
@@ -84,7 +84,14 @@ def refund_payment(kb_account_id, kb_payment_id, kb_payment_transaction_id, kb_p
       end
 
       def get_payment_info(kb_account_id, kb_payment_id, properties, context)
-        t_info_plugins = super(kb_account_id, kb_payment_id, properties, context)
+        ignored_api_calls = [:details_for]
```
Comment (pierre) [inline lib/paypal_express/api.rb]: I suppose this is to deal with existing 'bad' production data?

File: lib/paypal_express/api.rb [hunk 2]
```diff
@@ -288,7 +295,8 @@ def get_payer_info(token,
         payment_processor_account_id = payment_processor_account_id || :default
         gateway                      = lookup_gateway(payment_processor_account_id, kb_tenant_id)
         gw_response                  = gateway.details_for(token)
-        response, transaction        = save_response_and_transaction(gw_response, :details_for, kb_account_id, kb_tenant_id, payment_processor_account_id, kb_payment_id, kb_payment_transaction_id, transaction_type)
+
+        response = @response_model.create_response(:details_for, kb_account_id, kb_payment_id, kb_payment_transaction_id, transaction_type, payment_processor_account_id, kb_tenant_id, gw_response)
```
Comment (daliwei) [inline lib/paypal_express/api.rb]: Do we need to save the `details_for` response with `payment` and `transaction` id? If not, then the changes at `get_payment_info` can be avoided. The `details_for`'s response won't be returned as part of the `get_payment_info`

Comment (alenad) [inline lib/paypal_express/api.rb]: These are details for that payment and transaction, so we need to link them correctly.

**Comment by daliwei** [comment]
👍

**Pull request merged**